### PR TITLE
Convert to TypeScript

### PR DESCRIPTION
**Pull request by scribetw**

### CODE REVIEW

File: zk/src/archive/web/js/zk/au.ts [hunk 1]
```diff
@@ -1516,6 +1532,8 @@ zAu.cmd0 = /*prototype*/ { //no uuid at all
 				var w = Widget.$(uuid),
 					efs = w ? w.effects_ : null;
 				if (efs && efs.showBusy) {
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
```
Comment (scribetw) [inline zk/src/archive/web/js/zk/au.ts:1536]: can be removed

File: zk/src/archive/web/js/zk/dom.ts [hunk 2]
```diff
@@ -135,23 +139,23 @@ zjq = function (jq) { //ZK extension
 	}
 
 	
-	function _cmOffset(el) {
+	function _cmOffset(el: HTMLElement | null): zk.Offset {
```
Comment (scribetw) [inline zk/src/archive/web/js/zk/dom.ts:142]: strip null

File: zk/src/archive/web/js/zk/dom.ts [hunk 3]
```diff
@@ -170,31 +174,31 @@ zjq = function (jq) { //ZK extension
...
-					_ofsParent(el) : el.offsetParent;
+					_ofsParent(el) : el.offsetParent as HTMLElement;
 			}
-		} while (el);
+		}
 		return [l, t];
 	}
-	function _posOffset(el) {
-		if (zk.webkit && jq.nodeName(el, 'tr') && el.cells.length)
+	function _posOffset(el: HTMLElement | null): zk.Offset {
```
Comment (scribetw) [inline zk/src/archive/web/js/zk/dom.ts:182]: strip null

File: zk/src/archive/web/js/zk/dom.ts [hunk 4]
```diff
@@ -550,11 +557,13 @@ zk.override(jq.fn, _jq, /*prototype*/ {
...
 });
-jq.fn.init.prototype = jq.fn;
+jq.fn['init'].prototype = jq.fn;
 
 jq.each(['remove', 'empty', 'show', 'hide'], function (i, nm) {
 	_jq[nm] = jq.fn[nm];
 	jq.fn[nm] = function () {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
 		return !this.selector && this[0] === document ? this : _jq[nm].apply(this, arguments);
```
Comment (scribetw) [inline zk/src/archive/web/js/zk/dom.ts:567]: ?

File: zk/src/archive/web/js/zk/dom.ts [hunk 5]
```diff
@@ -1556,18 +1574,18 @@ jq(el).zk.center(); //same as 'center'
 	 * @see #scrollOffset
 	 */
 	viewportOffset: function () {
```
Comment (scribetw) [inline zk/src/archive/web/js/zk/dom.ts:1576]: review starts here

File: zk/src/archive/web/js/zk/widget.ts [hunk 6]
```diff
@@ -2102,9 +2111,9 @@ wgt.$f().main.setTitle("foo");
 	 */
 	setTopmost: function () {
 		if (!this.desktop) return -1;
-		if (this._userZIndex) return;
+		if (this._userZIndex) return this._zIndex;
```
Comment (scribetw) [inline zk/src/archive/web/js/zk/widget.ts:2114]: return -1 or this._zIndex?

File: zk/src/archive/web/js/zk/widget.ts [hunk 7]
```diff
@@ -2396,7 +2409,6 @@ redraw: function (out) {
 			if (!s.height && this._vflex)
 				s.height = n.style.height;
 			zk(n).clearStyles().jq.css(s);
-
```
Comment (scribetw) [inline zk/src/archive/web/js/zk/widget.ts:2399]: revert

File: zk/src/archive/web/js/zk/widget.ts (hunk 6 above)
Comment (scribetw) [reply zk/src/archive/web/js/zk/widget.ts:2114]:
```
var topZIndex = this.setTopmost();
pp.style.zIndex = topZIndex > 0 ? topZIndex : 1;
```
It's safer to use -1 (like the old result of undefined)

**Review by DevChu — APPROVED** [review]

**Review by plijyfes — APPROVED** [review]

**Pull request merged**